extensions/parquet/base/src/main/java/io/deephaven/parquet/base/BufferedStreamOverWriteChannel.java

```diff
@@ -0,0 +1,24 @@
+package io.deephaven.parquet.base;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.nio.channels.Channels;
+import java.nio.channels.SeekableByteChannel;
+
+final class BufferedStreamOverWriteChannel extends BufferedOutputStream {
+
+    private final SeekableByteChannel writeChannel;
+
+    BufferedStreamOverWriteChannel(final SeekableByteChannel writeChannel, final int size) {
```

Comment:
Please suggest a better name

Reply:
Interestingly there already exist a class `PositionOutputStream` that does exactly this. But the challenging part is that all of these are abstract classes and not interface (even `OutputStream` is abstract). And I need to extend from `BufferedOutputStream` since I need to access the `count` field. And I couldn't extend more than one abstract class.